espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ComposeNodeFinder.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.helpers
+
+import android.content.Context
+
+import androidx.test.espresso.DataInteraction
+import androidx.test.espresso.EspressoException
+import androidx.test.espresso.PerformException
+import android.view.View
+import android.widget.AdapterView
+import androidx.compose.ui.test.*
+
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+import org.hamcrest.TypeSafeMatcher
+
+import java.util.ArrayList
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidSelectorException
+import io.appium.espressoserver.lib.handlers.exceptions.XPathLookupException
+import io.appium.espressoserver.lib.model.Strategy
+import io.appium.espressoserver.lib.viewaction.ViewGetter
+
+import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.espresso.AppNotIdleException
+import androidx.test.espresso.Espresso.onData
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withClassName
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withTagValue
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import io.appium.espressoserver.EspressoServerRunnerTest
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.model.Locator
+import io.appium.espressoserver.lib.model.toJsonMatcher
+import io.appium.espressoserver.lib.viewmatcher.withView
+import io.appium.espressoserver.lib.viewmatcher.withXPath
+import org.hamcrest.CoreMatchers.allOf
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.Matchers.endsWith
+import org.hamcrest.Matchers.hasEntry
+import org.hamcrest.Matchers.instanceOf
+
+/**
+ * Helper methods to find elements based on locator strategies and selectors
+ */
+object ComposeNodeFinder {
```

Comment:
do we really need this singleton wrapper?  Kotlin allows high level function definitions

Reply:
added as high level function